hexrd/instrument/detector.py

```diff
@@ -1729,45 +1730,53 @@ def calc_physics_package_transmission(self, energy: np.floating,
     def calc_transmission_sample(self, seca: np.array,
                                  secb: np.array, energy: np.floating,
                                  physics_package: AbstractPhysicsPackage) -> np.array:
-        thickness_s = physics_package.sample_thickness # in microns
-        mu_s = 1./physics_package.sample_absorption_length(energy) # in microns^-1
+        thickness_s = physics_package.sample_thickness  # in microns
+        # in microns^-1
+        mu_s = 1./physics_package.sample_absorption_length(energy)
         x = (mu_s*thickness_s)
         pre = 1./x/(secb - seca)
         num = np.exp(-x*seca) - np.exp(-x*secb)
         return pre * num
 
     def calc_transmission_window(self, secb: np.array, energy: np.floating,
                                  physics_package: AbstractPhysicsPackage) -> np.array:
-        thickness_w = physics_package.window_thickness # in microns
-        mu_w = 1./physics_package.window_absorption_length(energy) # in microns^-1
+        thickness_w = physics_package.window_thickness  # in microns
+        # in microns^-1
+        mu_w = 1./physics_package.window_absorption_length(energy)
         return np.exp(-thickness_w*mu_w*secb)
 
     def calc_effective_pinhole_area(self, physics_package: AbstractPhysicsPackage) -> np.array:
         """get the effective pinhole area correction
         """
-        hod = (physics_package.pinhole_thickness /
-               physics_package.pinhole_diameter)
-        bvec = self.bvec
+        effective_pinhole_area = np.ones(self.shape)
 
-        tth, eta = self.pixel_angles()
-        angs = np.vstack((tth.flatten(), eta.flatten(),
-                  np.zeros(tth.flatten().shape))).T
-        dvecs = angles_to_dvec(angs, beam_vec=bvec)
+        if (physics_package.pinhole_diameter != 0.
+                and physics_package.pinhole_thickness != 0.):
```

Comment:
```suggestion
        if (not np.isclose(physics_package.pinhole_diameter, 0)
            and not np.isclose(physics_package.pinhole_thickness, 0)):
```

I think we should use `np.isclose()` for float comparisons here too.